### PR TITLE
enhanced artificial horizon

### PR DESCRIPTION
**Pull request by qubolino**
Brief summary of the changes
----------------------------
enhanced artificial horizon (square, with marks)

Related issues and discussions
------------------------------
not suitable for aerobatics
Screenshot 2023-07-05 at 18 09 50

### CODE REVIEW

File: src/Look/HorizonLook.cpp [hunk 1]
```diff
@@ -7,11 +7,17 @@
...
-  aircraft_pen.Create(Layout::Scale(2), COLOR_BLACK);
+  aircraft_pen.Create(Layout::Scale(1), Color(0xf2, 0xf2, 0x19));
+  aircraft_brush.Create(Color(0xf2, 0xf2, 0x19));
+  mark_pen.Create(Layout::Scale(1), COLOR_WHITE);
+  mark_brush.Create(COLOR_WHITE);
+  horizon_pen.Create(Layout::Scale(1), COLOR_WHITE);
 
   sky_brush.Create(sky_color);
-  sky_pen.Create(Layout::Scale(1), DarkColor(sky_color));
+  // sky_pen.Create(Layout::Scale(1), LightColor(sky_color));
```
Comment (lordfolken) [inline src/Look/HorizonLook.cpp:17]: No commented lines please.

File: src/Look/HorizonLook.hpp [hunk 2]
```diff
@@ -8,12 +8,17 @@
...
 struct HorizonLook {
   Pen aircraft_pen;
+  Brush aircraft_brush;
+  Pen mark_pen;
+  Brush mark_brush;
 
-  static constexpr Color sky_color{0x0a, 0xb9, 0xf3};
+  Pen horizon_pen;
+
+  static constexpr Color sky_color{0x13, 0x9c, 0xff};
```
Comment (lordfolken) [inline src/Look/HorizonLook.hpp:17]: Again history broken...

File: src/Renderer/HorizonRenderer.cpp [hunk 3]
```diff
@@ -41,36 +119,191 @@ HorizonRenderer::Draw(Canvas &canvas, const PixelRect &rc,
...
-  const int rr2p = uround(radius * M_SQRT1_2) + Layout::Scale(1);
-  const int rr2n = rr2p - Layout::Scale(2);
-  canvas.DrawLine({center.x + rr2p, center.y - rr2p},
-                  {center.x + rr2n, center.y - rr2n});
-  canvas.DrawLine({center.x - rr2p, center.y - rr2p},
-                  {center.x - rr2n, center.y - rr2n});
+  canvas.Select(look.aircraft_brush);
+  canvas.DrawLine({center.x + radius - radius/5, center.y}, {center.x + radius / 10, center.y});
+  canvas.DrawLine({center.x - radius + radius/5, center.y}, {center.x - radius / 10, center.y});
+  canvas.DrawLine({center.x + radius/10, center.y + radius / 10}, {center.x + radius/10, center.y});
```
Comment (lordfolken) [inline src/Renderer/HorizonRenderer.cpp:298]: Again broken history. if you move lines around then changes to them are not apparent.

File: src/Renderer/HorizonRenderer.cpp [hunk 4]
```diff
@@ -41,36 +119,191 @@ HorizonRenderer::Draw(Canvas &canvas, const PixelRect &rc,
...
+  canvas.DrawLine({center.x - radius/10, center.y + radius / 10}, {center.x - radius/10, center.y});
+  // canvas.DrawLine({center.x - 2, center.y}, {center.x +2, center.y});
```
Comment (lordfolken) [inline src/Renderer/HorizonRenderer.cpp:300]: commented line...

File: src/Look/HorizonLook.cpp [hunk 5]
```diff
@@ -7,11 +7,17 @@
...
+  sky_pen.Create(Layout::Scale(1), sky_color);
 
   terrain_brush.Create(terrain_color);
-  terrain_pen.Create(Layout::Scale(1), DarkColor(terrain_color));
+  // terrain_pen.Create(Layout::Scale(1), LightColor(terrain_color));
+  terrain_pen.Create(Layout::Scale(1), terrain_color);
```
Comment (lordfolken) [inline src/Look/HorizonLook.cpp:22]: Doing things this way, ruins the change view and makes it hard to review your commit.

**Review by lordfolken — CHANGES_REQUESTED** [review]
In general when you make commits then they should incremently change the code. 
Here you made changes, that you later commented or redid. 
* Your git history should be incremental steps to the final results.
* Every commit should compile. 
* Changes should be atomic.
* Commit messages need to be of the format: 

```
<component>: <change short version>
  longer text if needed. 
```

Comments on PRs are only for github. Git commits are forever.

**Comment by lordfolken** [comment]
Out of curiosity, why do you say its not for aerobatics?

**Comment by lordfolken** [comment]
Does it render well, when its in the infobox?

**Comment by lordfolken** [comment]
In general a big improvement of what was! Stick with it so we can get the code to a good standard.

**Comment by MaxKellermann** [comment]
> well i use git a bit differently and i imagine i should shrink all my commits into one to keep only incremental steps.

I don't know what that means, but certainly this PR should have more than one commit because it looks like this PR changes at least two independent things ("square" and "marks"). Each independent feature change must be one commit. Each refactoring change should be one commit.

Thanks @lordfolken, I agree that it is currently impossible to review this PR, it's a big mess.

**Pull request closed without merging**